apps/dotcom/client/src/utils/scratch-persistence-key.ts

```diff
@@ -9,11 +9,19 @@
  * users it does not need to be unique and we can just use a constant.
  */
 
-import { getFromLocalStorage, setInLocalStorage } from 'tldraw'
+import { getFromLocalStorage, setInLocalStorage, uniqueId } from 'tldraw'
 
 // DO NOT CHANGE THESE WITHOUT ADDING MIGRATION LOGIC. DOING SO WOULD WIPE ALL EXISTING LOCAL DATA.
 const defaultDocumentKey = 'TLDRAW_DEFAULT_DOCUMENT_NAME_v2'
 
-export const SCRATCH_PERSISTENCE_KEY =
-	getFromLocalStorage(defaultDocumentKey) ?? 'tldraw_document_v3'
-setInLocalStorage(defaultDocumentKey, SCRATCH_PERSISTENCE_KEY)
+export function getScratchPersistenceKey() {
+	return getFromLocalStorage(defaultDocumentKey) ?? 'tldraw_document_v3'
+}
+
+setInLocalStorage(defaultDocumentKey, getScratchPersistenceKey())
+
+export function resetScratchPersistenceKey() {
+	const id = 'tldraw_document_v3_' + uniqueId()
```

Comment:
Once consequence of this is that the user might delete local storage (but not indexed db) and loose access to their document. Before we would always default back to the same key, so that could not happen.

Reply:
alas you can't rename an indexeddb database so in order to avoid this we'd have to copy the contents to another db. Not impossible but meh if someone is messing around with their localstorage and they incur data loss it's their own fault IMO.

Reply:
Yeah, fair.